### PR TITLE
[air/docs] Add example to fetch results dataframe for trainer/tuner

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

Our docs examples should contain example code to fetch dataframes from result/result grid objects

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/trainer.rst [hunk 1]
```diff
@@ -212,8 +212,11 @@ You can interact with a `Result` object as follows:
     # returns the final metrics as reported
     result.metrics
 
-    # returns the contain an Exception if training failed.
+    # returns the Exception if training failed.
     result.error
 
+    # Returns a pandas dataframe of all reported results
+    results.metrics_dataframe
```
Comment (amogkam) [inline doc/source/ray-air/trainer.rst:219]: `ResultGrid` doesn't have a `metrics_dataframe` attribute. It's a `get_dataframe()` method instead. And `get_dataframe()` only returns the last result from each trial.

Comment (krfricke) [reply doc/source/ray-air/trainer.rst:219]: This is the `Result` class here though, right?

Comment (krfricke) [inline doc/source/ray-air/trainer.rst:219]:
```suggestion
    result.metrics_dataframe
```

File: doc/source/ray-air/doc_code/tuner.py [hunk 2]
```diff
@@ -201,7 +200,11 @@ def get_another_dataset():
 # And the best metrics
 best_metric = best_result.metrics
 
-# Inspect all results
+# Or a dataframe for further analysis
+results_df = best_result.metrics_dataframe
```
Comment (krfricke) [inline doc/source/ray-air/doc_code/tuner.py:204]:
```suggestion
results_df = results.get_dataframe()
```

File: doc/source/ray-air/trainer.rst (hunk 1 above)
Comment (krfricke) [reply doc/source/ray-air/trainer.rst]: Ah sorry, you meant the previous example. Fixed it

File: doc/source/ray-air/doc_code/tuner.py [hunk 3]
```diff
@@ -201,7 +200,11 @@ def get_another_dataset():
 # And the best metrics
 best_metric = best_result.metrics
 
-# Inspect all results
+# Or a dataframe for further analysis
+results_df = results.get_dataframe()
```
Comment (krfricke) [inline doc/source/ray-air/doc_code/tuner.py:204]:
```suggestion
results_df = result_grid.get_dataframe()
```

File: doc/source/ray-air/trainer.rst (hunk 1 above)
Comment (amogkam) [reply doc/source/ray-air/trainer.rst]: Ah yes that’s my bad!

**Review by amogkam — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**